docs/ref/csp.txt

```diff
@@ -0,0 +1,279 @@
+
+
+        @csp_exempt(enforced=False)
+        def another_view(request):
+            return HttpResponse("This view will not have an enforced CSP header")
+
+
+.. function:: csp_override(config, enforced=True, report_only=True)(view)
+
+    This decorator allows you to override the Content Security Policy for a specific view.
+    It takes a dictionary as an argument, similar to the ``SECURE_CSP`` and ``SECURE_CSP_REPORT_ONLY``
+    settings. This decorator also takes optional boolean arguments ``enforced`` and ``report_only``
+    to customize which header is overridden. By default, both headers are overridden. Example:
+
+    .. code-block:: python
+
+        from django.http import HttpResponse
+        from django.middleware import csp
+        from django.views.decorators.csp import csp_override
+
+
+        @csp_override(
+            {
+                "default-src": [csp.SELF],
+                "img-src": [csp.SELF, "data:"],
+            }
+        )
+        def my_view(request):
+            return HttpResponse("Both CSP headers will be overridden")
+
+
+        @csp_override(
+            {
+                "default-src": [csp.SELF],
+                "img-src": [csp.SELF, "data:"],
+            },
+            report_only=False,
+        )
+        def enforced_only_view(request):
+            return HttpResponse("Only the enforced CSP header will be overridden")
+
+
+        @csp_override(
+            {
+                "default-src": [csp.SELF],
+                "img-src": [csp.SELF, "data:"],
+            },
+            enforced=False,
+        )
+        def report_only_view(request):
+            return HttpResponse("Only the report-only CSP header will be overridden")
+
+These decorators provide fine-grained control over the Content Security Policy
+on a per-view basis, allowing you to customize the policy as needed for specific
+parts of your application.
+
+Limitations and Considerations
+==============================
+
+While Content Security Policy (CSP) is a powerful security feature, it's important to be aware of
+its limitations and considerations, especially in the context of Django's implementation:
+
+* Browser Support: While most modern browsers support CSP, support for directives vary.
+
+* Performance Impact: Implementing CSP, especially with nonces, can have a slight performance
+  impact due to the additional processing required for each request.
+
+* Third-party Scripts: If your application relies heavily on third-party scripts or services,
+  implementing a strict CSP can be challenging and may require careful configuration.
+
+* Report-Only Mode Limitations: While useful for testing, the report-only mode
+  (``SECURE_CSP_REPORT_ONLY``) doesn't actually prevent any violations, it only reports them.
+
+* Nonce Limitations: Nonces are single-use tokens. If you're caching pages that include nonces,
+  you'll need to implement additional logic to ensure the nonces are refreshed appropriately.
```

Comment:
Nit: My reading is that this isn't quite true. The main issue is that the nonce must match in both the header and cached content, rather than the token being only valid for a single request. If a HTML page serves a nonce, and uses it in a referenced JS file, it should be fine regardless of caching.

Reply:
Yes, the wording could be updated here. I think the _intention_ behind the nonce is single-use so I think it's worth mentioning potential considerations for caching, but to the browser as long as the nonce in the header matches the nonce in the HTML, it won't cause a CSP violation.

Reply:
I believe the same principle is true for the CSRF token - it's possible there's some wording from those docs you can use as a basis.

Reply:
I updated this to instead bring up the caching consideration and point out both header and content nonces need to match.